### ISSUE TITLE
OSError: [Errno 2] No such file or directory: '2018-02-26.p' on day change

### ISSUE THREAD

**Issue by vonpupp** [enigmampc/catalyst#253]
Dear Catalyst Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Operating System:
'Linux 4.4.0-98-generic #121-Ubuntu SMP Tue Oct 10 14:24:03 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
* Python Version:
`Python 2.7.12`
* Python Bitness: `64`
* How did you install Catalyst: `pip`
* Python packages: `$ pip freeze`
```
alembic==0.9.8
backports.shutil-get-terminal-size==1.0.0
bcolz==0.12.1
Bottleneck==1.2.1
ccxt==1.10.1187
certifi==2018.1.18
chardet==3.0.4
click==6.7
contextlib2==0.5.5
cyordereddict==1.0.0
Cython==0.27.3
decorator==4.2.1
empyrical==0.2.1
-e git+https://github.com/enigmampc/catalyst.git@388535b09c64c95f565397b2736b8576a7c73cab#egg=enigma_catalyst
enum34==1.1.6
future==0.16.0
idna==2.6
intervaltree==2.1.0
ipdb==0.11
ipython==5.5.0
ipython-genutils==0.2.0
Logbook==1.2.1
lru-dict==1.1.6
lxml==4.1.1
Mako==1.0.7
MarkupSafe==1.0
multipledispatch==0.4.9
networkx==2.1
numexpr==2.6.4
numpy==1.14.1
pandas==0.19.2
pandas-datareader==0.6.0
pathlib2==2.3.0
patsy==0.5.0
pexpect==4.4.0
pickleshare==0.7.4
pkg-resources==0.0.0
prompt-toolkit==1.0.15
ptyprocess==0.5.2
Pygments==2.2.0
python-dateutil==2.6.1
python-editor==1.0.3
python-telegram-bot==9.0.0
pytz==2018.3
redo==1.6
requests==2.18.4
requests-file==1.4.3
requests-ftp==0.3.1
requests-toolbelt==0.8.0
scandir==1.7
scipy==1.0.0
simplegeneric==0.8.1
six==1.11.0
sortedcontainers==1.5.9
SQLAlchemy==1.2.4
statsmodels==0.8.0
TA-Lib==0.4.16
tables==3.4.2
toolz==0.9.0
traitlets==4.3.2
transitions==0.6.4
urllib3==1.22
wcwidth==0.1.7
wrapt==1.10.11
```

Now that you know a little about me, let me tell you about the issue I am having:

# Description of Issue

When I run an algo on `paper mode` with `1H` or `2H` resampling it breaks `after midnight` (UTC) when the day changes:

```
period_close                                                                                                                    
2018-02-25 23:58:00+00:00        1111.11      1111.11          1111.11  0.0              0               0       0             0
[2018-02-25 23:58:00.154457] INFO: exchange_algorithm: portfolio balances, cash: 1111.11, positions: 0.0
[2018-02-25 23:58:00.155904] INFO: d_papert_abba_brittany: IGNORE 3/388 (1H): 2018-02-25 23:58:00+00:00
[2018-02-25 23:58:00.223172] INFO: exchange_algorithm: statistics for the last 1 minutes:
                           starting_cash  ending_cash  portfolio_value  pnl  long_exposure  short_exposure  orders  transactions
period_close                                                                                                                    
2018-02-25 23:59:00+00:00        1111.11      1111.11          1111.11  0.0              0               0       0             0
[2018-02-25 23:59:00.402624] INFO: exchange_algorithm: portfolio balances, cash: 1111.11, positions: 0.0
[2018-02-25 23:59:00.403042] INFO: d_papert_abba_brittany: IGNORE 3/389 (1H): 2018-02-25 23:59:00+00:00
[2018-02-25 23:59:00.506612] INFO: exchange_algorithm: statistics for the last 1 minutes:
                           starting_cash  ending_cash  portfolio_value  pnl  long_exposure  short_exposure  orders  transactions
period_close                                                                                                                    
2018-02-26 00:00:00+00:00        1111.11      1111.11          1111.11  0.0              0               0       0             0
Traceback (most recent call last):
  File "d_papert_abba_brittany.py", line 243, in <module>
    live_graph=False
  File "/home/av/repos/catalyst/catalyst/utils/run_algo.py", line 549, in run_algorithm
    stats_output=stats_output
  File "/home/av/repos/catalyst/catalyst/utils/run_algo.py", line 331, in _run
    overwrite_sim_params=False,
  File "/home/av/repos/catalyst/catalyst/exchange/exchange_algorithm.py", line 311, in run
    data, overwrite_sim_params
  File "/home/av/repos/catalyst/catalyst/algorithm.py", line 724, in run
    for perf in self.get_generator():
  File "/home/av/repos/catalyst/catalyst/gens/tradesimulation.py", line 224, in transform
    for capital_change_packet in every_bar(dt):
  File "/home/av/repos/catalyst/catalyst/gens/tradesimulation.py", line 137, in every_bar
    handle_data(algo, current_data, dt_to_use)
  File "/home/av/repos/catalyst/catalyst/utils/events.py", line 216, in handle_data
    dt,
  File "/home/av/repos/catalyst/catalyst/utils/events.py", line 235, in handle_data
    self.callback(context, data)
  File "/home/av/repos/catalyst/catalyst/exchange/exchange_algorithm.py", line 792, in handle_data
    self.nullify_frame_stats(now=data.current_dt)
  File "/home/av/repos/catalyst/catalyst/exchange/exchange_algorithm.py", line 766, in nullify_frame_stats
    rel_path='frame_stats'
  File "/home/av/repos/catalyst/catalyst/exchange/utils/exchange_utils.py", line 445, in remove_old_files
    creation_unix = os.path.getctime(f)
  File "/home/av/repos/example/.envc-local/lib/python2.7/genericpath.py", line 72, in getctime
    return os.stat(filename).st_ctime
OSError: [Errno 2] No such file or directory: '2018-02-26.p'
```

* What did you expect to happen? I expected to work normally.
* What happened instead? It breaks, this seems to happen on any timeframe greater or equal to 1H.

Here is how you can reproduce this issue on your machine:

## Reproduction Steps

1. Create a simple algo on paper mode
2. Use 1H or 2H resample
3. Wait

## What steps have you taken to resolve this already?

Nothing really, I think it is something internal on Catalyst.

Sincerely,
V.

**Comment by zackgow** [comment]
This is identical to my issue at #252 . It occurred at an identical time, during the date change.

**Comment by AvishaiW** [comment]
fixed on develop.

**Comment by vonpupp** [comment]
Awesome @AvishaiW, thank you very much. I will try that right away.

**Comment by AvishaiW** [comment]
because of the similarity to #252 , I'm closing this issue. there was another bug that I fixed- tagging only #252, so please make sure to update it, before running.
